xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -851,9 +852,35 @@ bool CGUIMediaWindow::Update(const std::string &strDirectory, bool updateFilterP
     }
   }
 
-  // if we haven't found the selected item, select the first item
+  // if we haven't found the selected item, see if we should select the first unplayed item
+  // if yes, find the first unplayed item and select it
+  // if no, select the first item
   if (!bSelectedFound)
-    m_viewControl.SetSelectedItem(0);
+  {
+    int iIndex = 0; // index of the item to select, default to the first item
+
+    // Check if we should select the first unplayed item
```

Comment:
It was meant to be generic, at the moment it's limited to tv show seasons/episodes but it could also be expanded out to music etc.

I just don't have music in my library to work with at the moment.

Still very new to the code, but with JumpToFirstUnplayedItem implemented at the CGUIViewState level any view state that inherits from that could override the method and enable the functionality, as long as CGUIMediaWindow is also involved somewhere :)